bin/src/commands.rs

```diff
@@ -20,3 +26,41 @@ pub async fn ui_error(message: String, _stack: Option<Vec<String>>) -> AppResult
 
     Ok(())
 }
+
+#[tauri::command]
+pub async fn add_contract(
+    chain_id: u64,
+    address: Address,
+    db: tauri::State<'_, Db>,
+) -> AppResult<()> {
+    let provider = provider(chain_id as u32).await.unwrap();
+    let proxy = ethui_proxy_detect::detect_proxy(address, &provider)
+        .await
+        .unwrap()
+        .map(|_| address);
```

Comment:
isn't this just replacing whatever value comes with the address of the proxy itself?
you're using it down there are the `proxy_for` argument, which should be the implementation